src/util/ConstexprUtils.h

```diff
@@ -199,4 +192,36 @@ auto cartesianPowerAsIntegerArray() {
   return toIntegerSequence<cartesianPowerAsArray<Upper, Num>()>();
 }
 
+/*
+@brief Call the given lambda function with each of the given types `Ts` as
+explicit template parameter, keeping the same order.
+*/
+template <typename... Ts>
+constexpr void constExprForEachType(const auto& lambda) {
+  (lambda.template operator()<Ts>(), ...);
+}
+
+/*
+@brief Call the given lambda function with each type in the given `std::variant` type as explicit
+template parameter, keeping the same order.
+*/
+template <typename VariantType>
+requires isVariant<VariantType> constexpr void constExprForEachTypeInVariant(const auto& lambda) {
+  ConstexprForLoop(
+      std::make_index_sequence<std::variant_size_v<VariantType>>{},
+      [&lambda]<size_t index,
+                typename IndexType = std::variant_alternative_t<index, VariantType>>() {
+        lambda.template operator()<IndexType>();
+      });
+}
+
+/*
+@brief Call the given lambda function with each type in the given `std::tuple` type as explicit
+template parameter, keeping the same order.
+*/
+template <typename TupleType>
+requires isTuple<TupleType> constexpr void constExprForEachTypeInTuple(const auto& lambda) {
+  constExprForEachTypeInVariant<TupleToVariant<TupleType>>(lambda);
+}
+
```

Comment:
This can be implemented generically for all templates that only take type parameters, see
[here](https://godbolt.org/z/7MecMqsof) for a small example (the trick is partial specialization for th etemplate pattern matching.

Reply:
maybe you can then call it `forEachTypeInTemplate<std::tuple<int, bool, string_view>>(f)`

Reply:
That is a very good idea. Done in [1a7d360](https://github.com/ad-freiburg/qlever/pull/1147/commits/1a7d36058a00866e3032d16c0aa8bb1e7465134f).